1introduction/tutorial.md

```diff
@@ -0,0 +1,205 @@
+
+  ```swift
+  import UIKit
+
+  class ViewController: UIViewController {
+
+
+  }```
+
+7. Hold down the **ctrl** button on your keyboard, then click and drag from the button in your storyboard into the `ViewController` code - on an empty line between the `{  ... }`
+
+  In the pop-up that appears:
+  * Set the **Connection Type** to be `Action`
+  * Type the **Name** of the function as `buttonTapped`
+
+  Hit **Connect**
+
+  ![step7](./assets/step7.gif)
+
+  This has created an `IBAction` - which is an function in our code that is triggered by an action in our storyboard.
+
+8. Let's see if our button works!
+
+  First, in the body of the `buttonTapped` function let's add a print statement:
+
+  ```swift
+  @IBAction func buttonTapped(_ sender: Any) {
+      print("Hello, world")
+  }
+  ```
+
+9. Run your app in the iPhone 7 Plus simulator again, and tap on the button.
+
+  ![step8](./assets/step8.gif)
+
+  You should see your message appear in the console!
+
+  /giphy Celebrate
+
+### Count the Clicker
+
+10. Time to build our clicker.
+
+  As you did with the button, find a `Label` from the `Object Library` and drag it onto the storyboard, in the middle of the view.
+
+  (Check back to Step 2 of the last section for a reminder of how to do this)
+
+11. The text of the label is a little small. Head over to **Attributes Inspector** in the **Utility Area** (right hand pane).
+
+  ![step8](./assets/step8.png)
+
+  Increase the font size to 100pt. The original size of the label is now laughably small for our epic font size, so now drag the corners to make it big enough.
+
+  ![step9](./assets/step9.gif)
+
+12. Set the text alignment of your label to be `Centered` - it's in the same place as you set the font size.
+
+13. Now, let's create a link between our label and the `ViewController`.
+
+  We do this using an `IBOutlet`, and it's created the same way as the `IBAction` we used to link our button to the `ViewController`.
+
+  Give it a try - look back at Step 7 if you need to.
+
+  When the popup appears:
+    * Set the **Connection Type** to `Outlet`
+    * Give it the name `counterLabel`
+
+  Your `ViewController` should now look like this:
+
+  ```swift
+    import UIKit
+
+    class ViewController: UIViewController {
+
+      @IBOutlet weak var counterLabel: UILabel!
+
+      @IBAction func buttonTapped(_ sender: Any) {
+        print("Hello, world")
+      }
+
+    }
+  ```
+
+14. Next we want to get actions on our button changing the appearance of our label.
+
+  Inside of your `buttonTapped` function add the following line:
+
+  `counterLabel.text = "🙌"`
+
+  (Hint: To bring up the emoji keyboard, hit **ctrl** + **cmd** + **space**)
+
+  Run your app, and hit that button, and (fingers crossed) see your emoji appear.
+
+### 3 ... 2 ... 1 ... 🚀
+
+15. Finally, let's make the counter increment (good bye emojis, for now).
+
+  For this, our ViewController is going to need a local variable. Add this line to the top of your `ViewController`:
```

Comment:
Would "property", "instance variable" or just "variable" be better than "local variable" as strictly it isn't a local variable? Sorry to be picky.